store/tikv/region_cache.go

```diff
@@ -416,6 +416,8 @@ func (c *RegionCache) OnRequestFail(ctx *RPCContext) {
 	delete(c.storeMu.stores, storeID)
 	c.storeMu.Unlock()
 
+	log.Warnf("drop regions from cache due to request fail, storeID: %v", storeID)
```

Comment:
For leader change or region split, server will response a NotLeader or RegionNotFound. So just network timeout will cause request fail. Any second thoughts? cc @tiancaiamao